sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdResponseHeaders.java

```diff
@@ -296,217 +268,114 @@ public String toString() {
-            toIntegerEntry(BackendHeaders.COLLECTION_SERVICE_INDEX, token)
-        );
-
-        collector.accept(this.collectionUpdateProgress, token ->
-            toIntegerEntry(HttpHeaders.COLLECTION_INDEX_TRANSFORMATION_PROGRESS, token)
-        );
-
-        collector.accept(this.continuationToken, token ->
-            toStringEntry(HttpHeaders.CONTINUATION, token)
-        );
-
-        collector.accept(this.currentReplicaSetSize, token ->
-            toIntegerEntry(BackendHeaders.CURRENT_REPLICA_SET_SIZE, token)
-        );
-
-        collector.accept(this.currentWriteQuorum, token ->
-            toIntegerEntry(BackendHeaders.CURRENT_WRITE_QUORUM, token)
-        );
-
-        collector.accept(this.databaseAccountId, token ->
-            toStringEntry(BackendHeaders.DATABASE_ACCOUNT_ID, token)
-        );
-
-        collector.accept(this.disableRntbdChannel, token ->
-            toBooleanEntry(HttpHeaders.DISABLE_RNTBD_CHANNEL, token)
-        );
-
-        collector.accept(this.eTag, token ->
-            toStringEntry(HttpHeaders.E_TAG, token)
-        );
-
-        collector.accept(this.globalCommittedLSN, token ->
-            toLongEntry(BackendHeaders.GLOBAL_COMMITTED_LSN, token)
-        );
-
-        collector.accept(this.hasTentativeWrites, token ->
-            toBooleanEntry(BackendHeaders.HAS_TENTATIVE_WRITES, token)
-        );
-
-        collector.accept(this.indexingDirective, token ->
-            new Entry(HttpHeaders.INDEXING_DIRECTIVE, RntbdIndexingDirective.fromId(token.getValue(Byte.class)).name())
-        );
-
-        collector.accept(this.isRUPerMinuteUsed, token ->
-            toByteEntry(BackendHeaders.IS_RU_PER_MINUTE_USED, token)
-        );
-
-        collector.accept(this.itemCount, token ->
-            toIntegerEntry(HttpHeaders.ITEM_COUNT, token)
-        );
-
-        collector.accept(this.itemLSN, token ->
-            toLongEntry(BackendHeaders.ITEM_LSN, token)
-        );
-
-        collector.accept(this.itemLocalLSN, token ->
-            toLongEntry(BackendHeaders.ITEM_LOCAL_LSN, token)
-        );
-
-        collector.accept(this.lastStateChangeDateTime, token ->
-            toStringEntry(HttpHeaders.LAST_STATE_CHANGE_UTC, token)
-        );
-
-        collector.accept(this.localLSN, token ->
-            toLongEntry(BackendHeaders.LOCAL_LSN, token)
-        );
-
-        collector.accept(this.logResults, token ->
-            toStringEntry(HttpHeaders.LOG_RESULTS, token)
-        );
-
-        collector.accept(this.numberOfReadRegions, token ->
-            toIntegerEntry(BackendHeaders.NUMBER_OF_READ_REGIONS, token)
-        );
-
-        collector.accept(this.offerReplacePending, token ->
-            toBooleanEntry(BackendHeaders.OFFER_REPLACE_PENDING, token)
-        );
+    private Map<String, String> collectEntries(final RntbdContext context, final UUID activityId) {
+        Map<String, String> responseHeaders = new HashMap<>(this.computeCount() + 2);
```

Comment:
I remember I tested this a few months back and I found out that guava ImmutableMap construction is more performant than HashMap construction.
in my old perf testing replacing ImmutableMap with HashMap resulted in perf drop.

Could you please do a fully e2e perf testing to compare the perf difference of this PR?